### PR TITLE
docs(root): added icon button variants with web component and React c…

### PR DESCRIPTION
**Pull request by GCHQDeveloper823**
…ode snippets

Added icon button variants with web component and React code snippets

## Summary of the changes

Added icon button variants (primary, secondary, tertiary, destructive) with Web Component and React code snippets

## Related issue
#1010

## Checklist

- [] I have [manually accessibility tested](https://design.sis.gov.uk/accessibility/testing/manual-testing) any changes, if relevant.
- [X] I have raised this pull request against the [develop branch](https://github.com/mi6/ic-design-system/tree/develop)

### CODE REVIEW

**Review by GCHQ-Developer-299 — DISMISSED** [review]

**Review by GCHQ-Developer-112 — CHANGES_REQUESTED** [review]
Putting this on hold at the moment until the StackBlitz work goes in. Hoping this will be unblocked by tomorrow

**Review by gd2910 — DISMISSED** [review]

**Comment by GCHQ-Developer-112** [comment]
> Putting this on hold at the moment until the StackBlitz work goes in. Hoping this will be unblocked by tomorrow

This is unblocked now. Please rebase off origin/develop and follow the new format for code snippets. Thank you!

**Comment by GCHQ-Developer-530** [comment]
Noticed the web components and react code snippets still have one button with `variant="icon"` but we're not showing an example of that in the component preview. Not sure which is best, but we'll either want to remove the code snippets for that or add another button into the component preview.

File: src/content/structured/components/buttons/code.mdx [hunk 1]
```diff
@@ -346,22 +392,86 @@ export const snippetsIcon = [
...
+    fill="#000000"
+  >
     <path d="M0 0h24v24H0V0z" fill="none"></path>
     <path d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"></path>
   </svg>
 </IcButton>
 <IcButton
   variant="icon"
   aria-label="you can disable tooltips on icon buttons"
-  disableTooltip
```
Comment (GCHQ-Developer-112) [inline src/content/structured/components/buttons/code.mdx:364]: Why was this removed?

Comment (GCHQDeveloper823) [reply src/content/structured/components/buttons/code.mdx]: Added this back in to the icon variant 👍

**Review by GCHQ-Developer-530 — APPROVED** [review]

**Review by GCHQ-Developer-299 — APPROVED** [review]
